### ISSUE TITLE
How can I get the output word embeddings?

### ISSUE THREAD

**Issue by ghost** [kpot/keras-transformer#4]
I trained a dataset using the vanilla transformer, but the .h5 file does not have the output word embeddings generated by the model. Is there any way I can access that?

Also, how can I use pre-trained word vectors instead of the BPE encoding to start with?

**Comment by kpot** [comment]
Hi!

> I trained a dataset using the vanilla transformer, but the .h5 file does not have the output word embeddings generated by the model. Is there any way I can access that?

First, all parameters are there, you just missed them. Sorry, but I can't go here into details on how Keras structures the HDF file while saving the model. I suggest you read Keras source code to figure that out, or just inspect the HDF file more carefully, looking for anything containing `bpe_embeddings` in its name.

Second, you need to define what you call the "output word embeddings". The way I understand it, it's embedding used before the output softmax layer. In this case, the model actually uses the same embedding matrix both at the input and the output. The trick is called "tied word embeddings". Check out the description of [`TiedOutputEmbedding`](https://github.com/kpot/keras-transformer/blob/master/keras_transformer/extras.py#L34) to learn more.

> Also, how can I use pre-trained word vectors instead of the BPE encoding to start with?

The code you're launching is not part of the library. Not "the main product" so to speak. It's just an example of how it can be used. Ability to initialize its model with various word embeddings, or export/import them, or make it extendable with plugins, etc - that would be already too much. You can modify it yourself, if you wish, and change tokenization from BPE to a "regular" word tokens. This would allow you to use any word embeddings your like for initialization.

**Comment by ghost** [comment]
Thank you so much for your detailed reply.
